### PR TITLE
[W4.4][T11-A4]Eka Buyung Lienadi

### PR DESCRIPTION
**Pull request by yungyung04**
Please kindly review.
Feed backs are really appreciated.
Thank you.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -9,14 +9,14 @@
...
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can only be in the specified format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
     private boolean isPrivate;
 
     /**
-     * Validates given address.
+     * Validates given address and
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:19]: and ??

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -9,14 +9,14 @@
 public class Address {
 
     public static final String EXAMPLE = "123, some street";
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:11]: Should update this attribute also.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -9,14 +9,14 @@
...
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can only be in the specified format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:15]: This attribute can be removed.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -27,15 +27,36 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        Street street = new Street(categories[1]);
+        Unit unit = new Unit(categories[2]);
+        PostalCode postalCode = new PostalCode(Integer.parseInt(categories[3]));
+        }
+        catch(ArrayIndexOutOfBoundsException e) {
+
+        }
+        catch(Exception exception){
+
+        }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:57]: You use try-catch with empty catch blocks, I don't think they're useful.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -27,15 +27,36 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     *  Partitions address into its block, street, unit and postal code.
+     *
+     *  Assumes valid input.
+     */
+    private void partitionAddress() {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:44]: Indentation problem inside this method.

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -27,15 +27,36 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        try {
+            String[] categories = value.split(",\\s", 4);
+        Block block = new Block(Integer.parseInt(categories[0]));
+        Street street = new Street(categories[1]);
+        Unit unit = new Unit(categories[2]);
+        PostalCode postalCode = new PostalCode(Integer.parseInt(categories[3]));
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java:50]: Can remove magic numbers. Based on the UML diagram, the new variables should be declared as class attributes.

File: src/seedu/addressbook/data/person/Block.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Block.java:3]: Missing header comment.

File: src/seedu/addressbook/data/person/PostalCode.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package seedu.addressbook.data.person;
+
+public class PostalCode {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/PostalCode.java:3]: Can implement a method to check if the postal code is in the correct format, e.g. number only.

**Comment by jeffryhartanto** [comment]
@yungyung04 Some comments added. Please close the PR after reading comments.

Overall, good job for doing this LO. Consider if any other method from the Address class can be implemented in the new classes. Moreover, you need to update the I/O test and documentation (UserGuide).

**Comment by yungyung04** [comment]
Thank you for the feed backs.

Do you mind if I ask some questions? 
Are we supposed to create a pull request for the other LOs that are not required to be evidence in our weekly outcome? Are we allowed to submit those LOs to request for a review?
Thank you.

File: src/seedu/addressbook/data/person/Address.java (hunk 6 above)
Comment (yungyung04) [reply src/seedu/addressbook/data/person/Address.java:50]: Can I generalize that when I have a composition relationship, I must set the objects instantiated from the 'part' class as class attribute of the 'whole' class? (I am referring to the prof's term to treat composition as "whole-part" relationship)

Comment (jeffryhartanto) [reply src/seedu/addressbook/data/person/Address.java:50]: Yes.. Good..

**Comment by jeffryhartanto** [comment]
You don't have to submit all of them, but you're always welcome to submit PRs for other LOs (e.g. Enum). However, due to time constraint, we usually review the "mandatory" or important PRs in detail, while skimming through other PRs. My suggestion for you is to write a specific question or doubt on the description of the PRs, so I can at least address that question/doubt.

**Pull request closed without merging**